extensions/search/README.md

```diff
@@ -0,0 +1,26 @@
+# Search Extension Specification
+
+**Extension [Maturity Classification](../../../extensions/README.md#extension-maturity): Proposal**
+
+This extension is intended to augment the core ItemCollection object when the ItemCollection is the result of a 
+search, for example, from calling the `/stac/search` API endpoint.
+
+- [Example](examples/example.json)
+- [JSON Schema](json-schema/schema.json)
+
+### ItemCollection fields
+
+| Element      | Type          | Description                                                  |
+| ------------ | ------------- | ------------------------------------------------------------ |
+| `search:metadata` | SearchMetadata Object | **REQUIRED.** The search-related metadata for the [ItemCollection](../itemcollection-spec/README.md). |
+
+### SearchMetadata Object
+
+| Element      | Type          | Description                                                  |
+| ------------ | ------------- | ------------------------------------------------------------ |
+| next         | string\|null  | **REQUIRED.** The value to set for the `next` query parameter in order to get the next page of results. |
+| returned     | integer       | **REQUIRED.** Count of results returned by this response. Equal to the cardinality of the features array. |
+| limit        | integer\|null | Max number of results requested explicitly or the default limit used by the server if no parameter was provided. `null` if no limit should be placed on the results, but servers may still limit the results nevertheless. |
```

Comment:
I tried to clarify this a bit as it seemed not very well described yet. Realized that when doing the JSON schemas. As it is not required, what is the default value? Is it set by the server implementation? Also added the note that servers may still limit the results for `null`.

Reply:
I'm going to change the language a bit.  This is for reporting back to the user what the `limit` value was if it existed (and null if there was no limit at all).  A few cases are:
* No value in the API call: limit = server default
* value in the API call larger than max limit:  limit = server max limit
* server has the (poor) behavior of no limit: limit = null

Reply:
Oh, my bad. I didn't realize that this is the response when writing the comment. Sorry.

Reply:
But if it's the response, a limit could always be delivered so why not make it required? The server should always know what its limit is and if not: null.

Reply:
I tend to agree.  I would even favor returning something like -1 if there were some circumstance where the server was incapable of knowing what limit was used on the back-end.  Returning null values just feels icky.

Reply:
Hah, I always feel bad when -1 is used and favor null for it. ;-)

Reply:
Lol... agree to disagree!  :)  I'm not going to fight for -1 vs null though.  If we want it to be null, then I guess we can all just do null checks!

Reply:
I put it this way because that's what it seemed other people wanted. I actually prefer the semantics where you just don't include the field if it would be null, so if next is not there, there are no more pages, and if limit is not there, there was no limit.

Reply:
For the openEO API we realized that it seems to be best to allow both: null and leaving it out. Both should have the same semantics. Non-compliant implementations often leave fields out and compliant back-ends often send null as it's sometimes easier to just set to null instead of removing a field again (at least where you have a relational database in the background, not so much in this case). I'd try to avoid different semantics for null and not set.